server/repositories/StakeholderRepository.js

```diff
@@ -22,8 +24,379 @@ class StakeholderRepository extends BaseRepository {
         [organization_id],
       );
 
+    // console.log('get by organization id ------> ', result.rows);
+
     return { stakeholders: result.rows, count: +count.rows[0].count };
   }
+
+  async getUUIDbyId(id) {
+    const stakeholder_id = await this._session
+      .getDB()(this._tableName)
+      .select('id')
+      .where('organization_id', id)
+      .first();
+
+    return stakeholder_id;
+  }
+
+  async verifyById(orgId, id) {
+    const stakeholder = await this._session
+      .getDB()(this._tableName)
+      .select('*')
+      .where('id', id)
+      .first();
+
+    return stakeholder;
+  }
+
+  async getAllStakeholderTrees(options) {
+    // get only non-children to start building trees
+    const results = await this._session
+      .getDB()('stakeholder as s')
+      .select('s.*')
+      .leftJoin('stakeholder_relations as sr', 's.id', 'sr.child_id')
+      .whereNull('sr.child_id')
+      // .orWhereNull('s.owner_id')
+      .orderBy('s.org_name', 'asc')
+      .limit(options.limit)
+      .offset(options.offset);
+
+    const stakeholders = await Promise.all(
+      results.map(async (stakeholder) => {
+        // eslint-disable-next-line no-param-reassign
+        stakeholder.parents = await this.getParents(stakeholder, options);
+        // eslint-disable-next-line no-param-reassign
+        stakeholder.children = await this.getChildren(stakeholder, options);
+        return stakeholder;
+      }),
+    );
+
+    const count = await this._session.getDB()('stakeholder as s').count('*');
+    // .leftJoin('stakeholder_relations as sr', 's.id', 'sr.child_id')
+    // .whereNull('sr.child_id');
+
+    return { stakeholders, count: +count[0].count };
+  }
+
+  async getAllStakeholderTreesById(id = null, options) {
+    // get only non-children to start building trees
+    const results = await this._session
+      .getDB()('stakeholder as s')
+      .select('s.*')
+      .leftJoin('stakeholder_relations as sr', 's.id', 'sr.child_id')
+      .where('s.id', id)
+      .orWhere('s.owner_id', id)
+      .andWhere('sr.child_id', null)
+      .orderBy('s.org_name', 'asc')
+      .limit(options.limit)
+      .offset(options.offset);
+
+    const stakeholders = await Promise.all(
+      results.map(async (stakeholder) => {
+        // eslint-disable-next-line no-param-reassign
+        stakeholder.parents = await this.getParents(stakeholder, options);
+        // eslint-disable-next-line no-param-reassign
+        stakeholder.children = await this.getChildren(stakeholder, options);
+        return stakeholder;
+      }),
+    );
+
+    // count all the stakeholder whether parent or child?
+    const count = await this._session
+      .getDB()(this._tableName)
+      .count('*')
+      .where('id', id)
+      .orWhere('owner_id', id);
+
+    return { stakeholders, count: +count[0].count };
+  }
+
+  // currently unused
+  async getStakeholderTreeById(id, options) {
+    const stakeholder = await this._session
```

Comment:
getStakeholderTreeById is more model logic than repository logic, since it's just a composition of repository calls.    This can be easily moved to the stakeholder model, right?